src/sparseml/core/logger/utils/frequency_manager.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import Union
+
+
+__all__ = ["FrequencyManager"]
+
+LogStepType = Union[int, float, None]
```

Comment:
Why do we call out int? Is there any int specific logic?

Reply:
Int will be needed for `optimizer_step` and `batch` frequency types; the support for that will come through in a different PR